### PR TITLE
[FIX] account_edi_ubl_cii: fix ubl import with allowance charge

### PR DESCRIPTION
**Pull request by aliyatastemirova**
When a UBL xml file is being uploaded and it contains an invoice line with an allowance charge, that charge is not being uploaded correctly. This commit:
- fixes the allowance charge xpath, so the node is found properly
- includes allowance charge amount in the discount calculation, because an allowance is not a discount.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_edi_ubl_cii/models/account_edi_common.py [hunk 1]
```diff
@@ -388,7 +388,7 @@ def _import_fill_invoice_allowance_charge(self, tree, invoice_form, journal, qty
         else:
             return
 
-        xpath = './{*}AllowanceCharge' if is_ubl else './{*}SupplyChainTradeTransaction/{*}ApplicableHeaderTradeSettlement/{*}SpecifiedTradeAllowanceCharge'
+        xpath = './{*}InvoiceLine/{*}AllowanceCharge' if is_ubl else './{*}SupplyChainTradeTransaction/{*}ApplicableHeaderTradeSettlement/{*}SpecifiedTradeAllowanceCharge'
```
Comment (JulienVR) [inline addons/account_edi_ubl_cii/models/account_edi_common.py:391]: The AllowanceCharges we are retrieving here are the document's one, the AllowanceCharges on the lines are retrieved in `_import_fill_invoice_line_values`. So I think there is a misunderstanding here.

File: addons/account_edi_ubl_cii/models/account_edi_common.py [hunk 2]
```diff
@@ -425,7 +425,7 @@ def _import_fill_invoice_allowance_charge(self, tree, invoice_form, journal, qty
                     invoice_line_form.price_unit = float(amount_node.text) * charge_factor * qty_factor
 
                 invoice_line_form.tax_ids.clear()  # clear the default taxes applied to the line
-                tax_xpath = './{*}TaxCategory/{*}Percent' if is_ubl else './{*}CategoryTradeTax/{*}RateApplicablePercent'
+                tax_xpath = '../{*}Item/{*}ClassifiedTaxCategory/{*}Percent' if is_ubl else './{*}CategoryTradeTax/{*}RateApplicablePercent'
```
Comment (JulienVR) [inline addons/account_edi_ubl_cii/models/account_edi_common.py:428]: same here, Item's are only present on the lines, it doesn't make sense on the document level

**Review by JulienVR — CHANGES_REQUESTED** [review]
Could you provide an example file ? :)

**Comment by smetl** [comment]
replaced by task: 3742296

**Pull request closed without merging**